### ISSUE TITLE
the UI screen used to configure app parameter values seems to have gone

### ISSUE THREAD

**Issue by rawlingsj** [fabric8io/fabric8#6319]
I just ran the hubot-slack app in a different namespace and expected to be taken to the page where I can override the default parameter values so I could enter the slack auth key.  Instead this page was skipped at the app was deployed with the default values.

**Comment by jstrachan** [comment]
ah that'll be my fault ;) we switched to using ConfigMaps so we don't use Templates by default.

I'll have a look at this one and try figure out a good approach. I'm not sure how many apps we have really need to be configured as we run them; but I guess hubot-slack is definitely one of them ;)

I figure we could annotate the ConfigMap's to provide Template-like UI while doing the same for real OpenShift Templates on OpenShift and for Helm charts too

**Comment by jstrachan** [comment]
apart from hubot-slack I wonder what other apps we should prompt users to configure. BTW you can configure the ConfigMap at any time now btw via the console

**Comment by rawlingsj** [comment]
Oh thats cool! I didn't know you could configure the config map in the console now.

I'll have a scan through to see which apps need configuring but yeah I doubt there's many.  I guess we might need to offer configmap or secrets to configure apps?

**Comment by jstrachan** [comment]
Yeah. We've added JSON Schema into the ConfigMaps as an annotation so you get a really nice UI to edit the configuration too ;)

**Comment by gashcrumb** [comment]
Just in here fixing another issue, we specifically disable the config form when not on openshift, I guess though there's no reason we couldn't pre-configure things when not on openshift, since we do the substitution in the console, is there?

**Comment by jstrachan** [comment]
yeah - we could just show the "Configure" UI (i.e. the ConfigMap page) when we Run stuff instead of the OpenShift Template UI

I guess at some point we might want to introduce 'install params' (what are the configuration parameters you have to specify in an OpenShift Template or in a Helm chart or should appear in the  UI in fabric8 console - which could be a smaller subset of the config page?)

Though for now showing the ConfigMap page would be very handy ;)

**Comment by gashcrumb** [comment]
This would be during the 'template' flow though, the configmap wouldn't likely be deployed yet.  Guess though we could dig it out of the objects to be deployed.  Sounds complicated :-)

**Comment by jstrachan** [comment]
Yeah ;) 

maybe for apps which we really need a few parameters to be set on install (i.e. like Template parameter names and values) - such as slack token for hubot-slack - we should iterate through all ConfigMaps and any with an annotation 'fabric8.io/parameters' we have a comma separated list of data keys? Then we show the same Template UI - it's just the values get put into one or more ConfigMaps rather than the Template.parameters?

**Comment by gashcrumb** [comment]
For templates we actually just substitute the values and submit the objects normally, i.e. there's really nothing openshift-specific in the way the template controller actually adds/updates objects from a template or config map, we just take the template parameters and do a search/replace to create the object and put/update it just like if you drag/drop a json/yaml file into the console.

So I guess then yeah, maybe it's just a matter of providing edit widgets for any config maps to be deployed.  Could get a bit painful though if for whatever reason a template happens to have lots of configmaps.

**Comment by jstrachan** [comment]
yeah - I'm hoping we can keep the UI side simple (one form) and just have a little bit of logic iterate through annotated ConfigMaps (rather than just Template.parameters)

**Comment by jstrachan** [comment]
Actually I think I'm going to fix this issue by just re-introducing Template objects back into the apps again - its the simplest thing that could work ;)

I love the idea of avoiding Templates for vanilla kubernetes & helm; then using them in openshift when required. So I'm gonna add the Templates back for the kubernetes ConfigMaps used for the Run... screen

**Comment by jstrachan** [comment]
I've been fixing this locally; bringing back template UIs where they make sense (e.g. hubot-slack in my local build). Its just gonna take until next week to get this all released...
